### PR TITLE
feat(chips): Create mixin to customize chip margins

### LINKED ISSUE #2265: Create mixin to customize chip margins
Create a mixin `mdc-chip-set-margins` to make the space between the chips is equal to the padding of the chip set.
image

Also in `mdc-chip-set` use `display: flex; flex-wrap: wrap;` to collapse the whitespace in the markup, because it will give us much better control over spacing (so we can match the spec).

### PR DESCRIPTION
**Pull request by bonniezhou**
Resolves #2265.
According to spec the margins should be 8px all around (in between chips and between edge of chip and edge of chip-set).

### CODE REVIEW

**Comment by bonniezhou** [comment]
The mixin is named `mdc-chip-set-margin` but I'm open to suggestions since "margin" made be misleading.

File: packages/mdc-chips/README.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ To customize the colors of any part of the chip, use the following mixins.
 
 Mixin | Description
 --- | ---
+`mdc-chip-set-margin($margin)` | Customizes the amount of space between each chip in the set
```
Comment (acdvorak) [inline packages/mdc-chips/README.md]: RE the word "margin": I think the design folks have a specific word for "the amount of space between stuff", but I can't remember what it is... "Gutter" maybe? I would check with them

File: packages/mdc-chips/chip-set/mdc-chip-set.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// limitations under the License.
+//
+
+@import "../mixins";
+
+.mdc-chip-set {
+  @include mdc-chip-set-margin(8px);
+
+  display: flex;
+  flex-wrap: wrap;
```
Comment (acdvorak) [inline packages/mdc-chips/chip-set/mdc-chip-set.scss]: Can we add `box-sizing: border-box` while we're here?

File: packages/mdc-chips/_mixins.scss [hunk 3]
```diff
@@ -87,3 +87,11 @@
   padding-left: $horizontal-padding-value;
   border-width: $width;
 }
+
+@mixin mdc-chip-set-margin($margin) {
+  padding: $margin / 2;
```
Comment (acdvorak) [inline packages/mdc-chips/_mixins.scss]: Nice! :grinning:

File: packages/mdc-chips/README.md (hunk 1 above)
Comment (kfranqueiro) [reply packages/mdc-chips/README.md]: I suspect gutter is more specific to e.g. layout grid and image list, things with a more consistent gap between rows/columns. Chips are inline and can wrap so I'm not sure it'd apply there, so better ask.

Comment (bonniezhou) [reply packages/mdc-chips/README.md]: Design suggested "padding" which I think makes more sense than "margin" but could be equally misleading. @acdvorak suggested "spacing", which sounds good to me.

Comment (acdvorak) [reply packages/mdc-chips/README.md]: It sounds like design wants "padding", but both "padding" and "margin" have well-established meanings on the Web, so using either of those would likely confuse a lot of our clients.

What about "spacing"?

**Pull request merged**